src/odb/src/lefout/lefout.cpp

```diff
@@ -135,32 +138,81 @@ void lefout::writeHeader(dbBlock* db_block)
   writeUnits(/*database_units = */db_block->getDbUnitsPerMicron());
 }
 
-void lefout::writeObstructions(dbBlock* db_block)
+void lefout::writeObstructions(dbBlock* db_block, int bloat_factor)
 {
-  std::set<dbTechLayer*> obstruction_layers;
-  getTechLayerObstructions(db_block, obstruction_layers);
+  std::map<dbTechLayer*,std::set<Rect*>> obstructions;
```

Comment:
Just use a std::set<Rect> and drop the indirection.  As written the code will leak all the Rect objects.

Reply:
I don't sure that I understand, can you elaborate?

Reply:
You calling 'new Rect' but never delete on those objects. 

However I think it would be best to drop Rect altogether and use polygon_90_set_data<int> here and directly add the shapes as I mentioned elsewhere.

Reply:
Got it, yes in that way I could merge the rects on the way and not in the end as you wrote below.

Reply:
Note that you could bloat the rects as you add them rather than bloating the whole set at the end.  The shapes will merge earlier that way potentially reducing memory usage.